### PR TITLE
feat(tag): add tag for public subnets

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
add discovery tag for public subnets - we want to use this subnets for cluster-autoscalers like karpenter, castai, clusterautoscaler

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

File: apis/basic/composition.yaml [hunk 1]
```diff
@@ -134,6 +141,8 @@ spec:
           patchSetName: network-id
         - type: PatchSet
           patchSetName: region
+        - type: PatchSet
+          patchSetName: network-id-tags
```
Comment (ytsarev) [inline apis/basic/composition.yaml:145]: nit: i would regroup and place this PatchSet next to `network-id`

File: apis/basic/composition.yaml [hunk 2]
```diff
@@ -31,6 +31,11 @@ spec:
         - type: FromCompositeFieldPath
           fromFieldPath: spec.parameters.region
           toFieldPath: spec.forProvider.region
+    - name: network-id-tags
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.parameters.id
+          toFieldPath: spec.forProvider.tags[networks.aws.platform.upbound.io/network-id]
```
Comment (ytsarev) [inline apis/basic/composition.yaml:38]: nit: i would regroup and place this PatchSet next to network-id

File: apis/basic/composition.yaml [hunk 3]
```diff
@@ -95,6 +100,8 @@ spec:
           patchSetName: network-id
         - type: PatchSet
           patchSetName: region
+        - type: PatchSet
+          patchSetName: network-id-tags
```
Comment (ytsarev) [inline apis/basic/composition.yaml:104]: nit: i would regroup and place this PatchSet next to network-id

**Review by ytsarev — APPROVED** [review]
Cosmetics suggestion, otherwise looks good

File: apis/basic/composition.yaml (hunk 1 above)
Comment (haarchri) [reply apis/basic/composition.yaml:143]: done

File: apis/basic/composition.yaml (hunk 3 above)
Comment (haarchri) [reply apis/basic/composition.yaml:102]: done

File: apis/basic/composition.yaml (hunk 2 above)
Comment (haarchri) [reply apis/basic/composition.yaml:33]: done

**Pull request merged**